src/main/java/io/cdap/plugin/batch/source/FTPBatchSource.java

```diff
@@ -317,5 +321,74 @@ private void validateAuthenticationInPath(FailureCollector collector) {
         collector.getOrThrowException();
       }
     }
+
+    /**
+     * This method checks if user enters a valid ftp path by first checking the server
+     * and port number, then the username and password. Will timeout after 10 seconds
+     * if connection is not established.
+     *
+     * @param collector
+     */
+    boolean getConnection(FailureCollector collector) {
```

Comment:
we're not checking the return value, just return void. In the unit test, check the failures in the mock collector

Reply:
see updates